_posts/2021-07-01-quarkus-2x-platform-quarkiverse-registry.adoc

```diff
@@ -0,0 +1,165 @@
+
+The Quarkus 2.0.0.Final release, in addition to the monolithic `io.quarkus:quarkus-universe-bom`, also includes the platform member-specific BOMs, e.g.
+
+* `io.quarkus.platform:quarkus-bom:2.0.0.Final` - an equivalent of the `io.quarkus:quarkus-bom`;
+* `io.quarkus.platform:quarkus-camel-bom:2.0.0.Final` - manages only the Camel Quarkus-related extensions and their required dependencies;
+* `io.quarkus.platform:quarkus-kogito-bom:2.0.0.Final` - manages only the Kogito Quarkus-related extensions and their required dependencies;
+* etc
+
+Having member-specific BOMs, applications now can import only the BOMs that are relevant to them and avoid enforcing dependency constraints from the rest of the "universe".
+
+NOTE: Given that every member-specific BOM is actually a fragment of the `io.quarkus:quarkus-universe-bom`, member BOMs can be imported in any order w/o creating a conflict.
+
+== Dev Tools Support
+
+The dev tools, such as the Quarkus CLI, Maven or Gradle plugins, can be used to create and manage a Quarkus application project using the new platform model.
+
+NOTE: The `io.quarkus:quarkus-universe-bom` is still the default BOM in Quarkus 2.0.0.Final but in one of the subsequent Quarkus 2.x releases it will be deprecated
+in favor of the member-specific BOMs.
+
+The easiest way to play with it would be using the new https://quarkus.io/guides/cli-tooling[Quarkus CLI], which can be installed using e.g.
+
+```shell
+curl -Ls https://sh.jbang.dev | bash -s - app install --fresh --force quarkus@quarkusio
+```
+
+Once installed you can create a project by executing
+
+```shell
+quarkus create app --registry-client
+```
+
+NOTE: Without the `--registry-client` argument the created project will be importing the default `io.quarkus:quarkus-universe-bom:2.0.0.Final`.
+
+The generated `pom.xml` will be importing the following BOM:
+```xml
+    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>2.0.0.Final</quarkus.platform.version>
+    <surefire-plugin.version>3.0.0-M5</surefire-plugin.version>
+  </properties>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>${quarkus.platform.group-id}</groupId>
+        <artifactId>${quarkus.platform.artifact-id}</artifactId>
+        <version>${quarkus.platform.version}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+```
+
+which is an equivalent of the `io.quarkus:quarkus-bom` that does not include any of the other platform members, such as Camel, Kogito, OptaPlanner, etc.
+
+However, if we did want to create a new project including a Kogito extension, it could be done like this:
+
+```shell
+quarkus create kogito-quarkus-rules --registry-client
```

Comment:
I think I'm a bit confused by the CLI, `kogito-quarkus-rules` is not the name of the project but a dependency?

Reply:
yes, it's an extension artifactId.

Reply:
For the record, I made the same assumption as @gsmet about expecting the project name in the CLi args

Reply:
right, it's a bit confusing

Reply:
@ebullient just fyi

Reply:
FYI, maven plugin been able to do this since day one too afaik. 

Just that our examples always verbose.

Reply:
btw. i would make the registry- client be first as the list of extension to look up is the last arguments.

```suggestion
quarkus create --registry-client kogito-quarkus-rules 
```

also just noticed the cli does not seem to lookup partial names as was done in past. ill open separate issue for that

Reply:
> FYI, maven plugin been able to do this since day one too afaik. 
> 
> Just that our examples always verbose. 

I don't think so. It needs -Dextensions=

Reply:
This is what the cli has always done (re: listing extensions).
Is it more natural to choose the artifactId (only) without an option, and require an option for additional extensions?

Reply:
@ebullient yes, I think it's more natural to choose the artifactId and require an option for additional extensions

Like: `quarkus create app foo --with-extensions prettytime quartz`

Reply:
yes, but i'm pretty sure people will want to specify the groupId instead of using the default one. Should we go for `quarkus create <GAV> --extensions xxx`?

Reply:
That would be a big change to make now (especially since this is how the CLI create path always worked.. free typing of extensions against default group/artifact/version). I mean.. the CLI wasn't advertised much, so you could argue that not many know it worked that way...

What you've shown is half-positional. `--with-extensions=one,two,three` or `-x one -x two -x three` would be more typical.

Reply:
See quarkusio/quarkus#18368 for updated create syntax